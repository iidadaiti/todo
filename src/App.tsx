import { customAlphabet } from 'nanoid'
import React, { FC, FormEvent, useState } from 'react'
import styles from './App.module.css'
import { Button } from './components/Button'
import { Header } from './components/Header'
import { InputText } from './components/InputText'
import { Page } from './components/Page'
import { Todo, TodoData, TodoProps } from './components/Todo'

const nanoid = customAlphabet('1234567890', 8)

export const App: FC = () => {
  const [todoList, setTodoList] = useState<TodoProps['todoList']>([])
  const [text, setText] = useState('')
  const addTodo = () => {
    const key = nanoid()
    const isDone = false
    const createAt = new Date()
    setTodoList([...todoList, { text, key, isDone, createAt }])
  }

  const handlerSubmit = (e: FormEvent) => {
    e.preventDefault()
    if (text.length < 1) {
      return
    }

    addTodo()
    setText('')
  }

  const handlerChangeTodo = (before: TodoData, after: TodoData) => {
    console.log('change')

    const replaceIndex = todoList.findIndex((it) => it === before)
    if (replaceIndex >= 0) {
      todoList[replaceIndex] = after
      setTodoList([...todoList])
    }
  }

  const handlerRemoveTodo = (todoData: TodoData) =>
    setTodoList(todoList.filter((it) => it !== todoData))

  return (
    <Page className={styles.page}>
      <header>
        <Header />
      </header>

      <main className={styles.main}>
        <section className={styles.section}>
          <div className={styles.container}>
            <div className={styles.todoHeader}>
              <form onSubmit={handlerSubmit}>
                <InputText
                  autoFocus
                  value={text}
                  onChange={(e) => setText(e.currentTarget.value)}
                />
                <div className={styles.todoHeaderButton}>
                  <Button type="submit">Add</Button>
                </div>
              </form>
            </div>

            <div className={styles.todoBody}>
              <Todo
                todoList={todoList}
                onChangeTodo={handlerChangeTodo}
                onRemoveTodo={handlerRemoveTodo}
              ></Todo>
            </div>
          </div>
        </section>
      </main>
    </Page>
  )
}
