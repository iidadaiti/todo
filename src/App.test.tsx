import { render } from '@testing-library/react'
import React from 'react'
import { App } from './App'

test('App snapshot', () => {
  const result = render(<App />)
  expect(result.asFragment()).toMatchSnapshot()
})
