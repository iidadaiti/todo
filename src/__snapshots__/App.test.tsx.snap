// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <h1
        class="title"
      >
        Todo
      </h1>
    </div>
    <input
      type="text"
      value=""
    />
    <button>
      Add
    </button>
    <div>
      <ul />
    </div>
  </div>
</DocumentFragment>
`;
